networks/2_module_validation.py

```diff
@@ -0,0 +1,300 @@
+
+for corr_threshold in corr_threshold_list:
+    print(f"Modules using correlation threshold: {corr_threshold}")
+    pao1_membership_filename = f"pao1_membership_{corr_threshold}.tsv"
+    pa14_membership_filename = f"pa14_membership_{corr_threshold}.tsv"
+
+    pao1_membership = pd.read_csv(
+        pao1_membership_filename, sep="\t", header=0, index_col=0
+    )
+    pa14_membership = pd.read_csv(
+        pa14_membership_filename, sep="\t", header=0, index_col=0
+    )
+
+    print(pao1_membership["module id"].value_counts())
+    print(pa14_membership["module id"].value_counts())
+
+
+# -
+
+# plotting function
+def plot_dist_modules(threshold_list):
+
+    # Set up the matplotlib figure
+    fig, axes = plt.subplots(ncols=2, nrows=3, figsize=(15, 15))
+    axes = axes.ravel()
+
+    for i in range(len(threshold_list)):
+        pao1_membership_filename = f"pao1_membership_{threshold_list[i]}.tsv"
+        pa14_membership_filename = f"pa14_membership_{threshold_list[i]}.tsv"
+
+        pao1_membership = pd.read_csv(
+            pao1_membership_filename, sep="\t", header=0, index_col=0
+        )
+        pa14_membership = pd.read_csv(
+            pa14_membership_filename, sep="\t", header=0, index_col=0
+        )
+
+        # Get bins using all data
+        hist, bins_corr = np.histogram(
+            np.concatenate([pao1_membership["module id"], pa14_membership["module id"]])
+        )
+
+        # Distribution plot for core genes
+        fig = sns.distplot(
+            pao1_membership["module id"],
+            label="PAO1 modules",
+            color="red",
+            bins=bins_corr,
+            kde=False,
+            ax=axes[i],
+        )
+
+        fig = sns.distplot(
+            pa14_membership["module id"],
+            label="PA14 modules",
+            color="blue",
+            bins=bins_corr,
+            kde=False,
+            ax=axes[i],
+        )
+
+        fig.set_title(
+            f"Histogram of size of modules using threshold {threshold_list[i]}",
+            fontsize=12,
+        )
+        handles, labels = fig.get_legend_handles_labels()
+        fig.legend(handles, labels, loc="center right")
+
+
+# Plot distribution of modules per threshold
+plot_dist_modules(corr_threshold_list)
+
+# **Takeaway:**
+# * Looks like as we decrease our correlation threshold (i.e. connections don't need to be very strong between nodes), more modules are able to form which is what I would expect.
+
+# ## Examine composition of modules
+#
+# We expect that genes within the same operon or regulon will cluster together (i.e. be within the same module). To test this we will compare the distribution of the number of modules that contain genes within the same regulon vs the number of modules that contain random genes
+
+# +
+# Load PAO1 regulon and operon file
+pao1_regulon_filename = paths.PAO1_REGULON
+pao1_operon_filename = paths.PAO1_OPERON
+
+# Load membership for specific threshold
+corr_threshold = 0.9
```

Comment:
Do these plots change much/at all if you vary the threshold?

Reply:
This trend held across varying thresholds: as we decreased thresholds, the KS stat for the operons became more significant while the KS stat for the regulons became slightly less signficant (0.08). The visual trend held despite differences in the p-value thought is is probably more due to the sample size differences since there are only 17 regulons compared to 3K operons